lib/tokens.js

```diff
@@ -449,6 +469,64 @@ export async function resolveToInternalId({token}) {
   return {internalId};
 }
 
+/**
+ * Sets the `minAssuranceForResolution` on the token batches associated
+ *   `internalId`.
+ *
+ * @param {object} options - Options to use.
+ * @param {string} options.internalId - The internal ID of the token batch.
+ * @param {number} options.minAssuranceForResolution - Minimum level of
+ *   assurance required for token resolution.
+ *
+ * @returns {Promise<boolean>} Returns true if update occurred.
+ */
+export async function setMinAssuranceForResolution({
+  internalId, minAssuranceForResolution
+}) {
+  const query = {
+    'tokenBatch.internalId': internalId,
+    'tokenBatch.remainingTokenCount': {$gt: 0}
+  };
+  const $set = {
+    'meta.updated': Date.now(),
+    'tokenBatch.minAssuranceForResolution': minAssuranceForResolution
+  };
+  const collection = database.collections['tokenization-tokenBatch'];
+  const result = await collection.updateMany(
+    query, {$set}, database.writeOptions);
+
+  // return `true` if the update occurred (existing document found)
+  return result.result.n !== 0;
+}
+
+/**
+ * Invalidates the token batch by marking it as invalid, but leaves
+ *   the record intact so previously resolved tokens are trackable. The
+ *   remaining token count is set to 0 so no new tokens can be created in the
+ *   batch.
+ *
+ * @param {object} options - Options to use.
+ * @param {string} options.internalId - The internal ID of the token batch(es).
+ *
+ * @returns {Promise<boolean>} Returns true if update occurred.
+ */
+export async function invalidateTokenBatch({internalId}) {
+  const query = {
+    'tokenBatch.internalId': internalId
```

Comment:
Hmm, I think I spoke too soon. We don't have an index on this and we can't add one. We're going to have to figure out what to do here.